cloudwatch/cloudwatch.go

```diff
@@ -756,7 +770,11 @@ func (output *OutputPlugin) putLogEvents(stream *logStream) error {
 			} else if awsErr.Code() == cloudwatchlogs.ErrCodeInvalidSequenceTokenException {
 				// sequence code is bad, grab the correct one and retry
 				parts := strings.Split(awsErr.Message(), " ")
-				stream.nextSequenceToken = &parts[len(parts)-1]
+				nextSequenceToken := &parts[len(parts)-1]
```

Comment:
can we add some comments here for explanation and also can we have the bug & the feature built in two commits?